File: server-test/src/test/java/de/njsm/stocks/servertest/v2/InitialServerTest.java
/*
 * stocks is client-server program to manage a household's food stock
 * Copyright (C) 2019  The stocks developers
 *
 * This file is part of the stocks program suite.
 *
 * stocks is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * stocks is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package de.njsm.stocks.servertest.v2;

import de.njsm.stocks.servertest.TestSuite;
import io.restassured.http.ContentType;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.when;
import static org.hamcrest.Matchers.*;

@Order(300)
public class InitialServerTest {

    @Test
    void foodIsEmpty() {
        when()
                .get(TestSuite.DOMAIN + "/v2/food").
        then()
                .log().ifValidationFails()
                .statusCode(200)
                .contentType(ContentType.JSON)
                .body("status", equalTo(0))
                .body("data", is(empty()));
    }

    @Test
    void locationsAreEmpty() {
        when()
                .get(TestSuite.DOMAIN + "/v2/location").
        then()
                .log().ifValidationFails()
                .statusCode(200)
                .contentType(ContentType.JSON)
                .body("status", equalTo(0))
                .body("data", is(empty()));
    }

    @Test
    void foodItemsAreEmpty() {
        when()
                .get(TestSuite.DOMAIN + "/v2/fooditem").
        then()
                .log().ifValidationFails()
                .statusCode(200)
                .contentType(ContentType.JSON)
                .body("status", equalTo(0))
                .body("data", is(empty()));
    }


    @Test
    void eansAreEmpty() {
        when().
                get(TestSuite.DOMAIN + "/v2/ean").
        then()
                .log().ifValidationFails()
                .statusCode(200)
                .contentType(ContentType.JSON)
                .body("status", equalTo(0))
                .body("data", is(empty()));
    }

    @Test
    void initialUserIsOnly() {
        when()
                .get(TestSuite.DOMAIN + "/v2/user").
        then()
                .log().ifValidationFails()
                .statusCode(200)
                .contentType(ContentType.JSON)
                .body("status", equalTo(0))
                .body("data", iterableWithSize(2));
    }

    @Test
    void initialDeviceIsOnly() {
        when()
                .get(TestSuite.DOMAIN + "/v2/device").
        then()
                .log().ifValidationFails()
                .statusCode(200)
                .contentType(ContentType.JSON)
                .body("status", equalTo(0))
                .body("data", iterableWithSize(2));
    }
}

File: server-test/src/test/java/de/njsm/stocks/servertest/v2/InitialServerTest.kt
/*
 * stocks is client-server program to manage a household's food stock
 * Copyright (C) 2019  The stocks developers
 *
 * This file is part of the stocks program suite.
 *
 * stocks is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * stocks is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */
package de.njsm.stocks.servertest.v2

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Order
import org.junit.jupiter.api.Test
import java.time.Instant

@Order(300)
class InitialServerTest : Base() {
    @BeforeEach
    fun setUp() {
        dagger.inject(this)
    }

    @Test
    fun foodIsEmpty() {
        assertThat(updateService.getFood(Instant.EPOCH))
            .isEmpty()
    }

    @Test
    fun locationsAreEmpty() {
        assertThat(updateService.getLocations(Instant.EPOCH))
            .isEmpty()
    }

    @Test
    fun foodItemsAreEmpty() {
        assertThat(updateService.getFoodItems(Instant.EPOCH))
            .isEmpty()
    }

    @Test
    fun eansAreEmpty() {
        assertThat(updateService.getEanNumbers(Instant.EPOCH))
            .isEmpty()
    }

    @Test
    fun initialUserAndSystemUserAreOnly() {
        assertThat(updateService.getUsers(Instant.EPOCH))
            .hasSize(2)
    }

    @Test
    fun initialDeviceAndSystemUserAreOnly() {
        assertThat(updateService.getUsers(Instant.EPOCH))
            .hasSize(2)
    }
}
